### DISCUSSION TITLE
nextjs-plugin-minify-classnames

### DISCUSSION THREAD

**Discussion by IRediTOTO** [vercel/next.js#17684] [Ideas]
Hi, I am trying to minify class name. Is there a package like this for Nextjs? https://github.com/stldo/gatsby-plugin-minify-classnames

**Comment by HaNdTriX** [comment]
What styling solution are you using?

**Reply by IRediTOTO** [reply]
I'm using CSS in JS emotion. But I would love a seperate CSS file load

**Comment by matamatanot** [comment]
Check #14197 out!

**Comment by kenmueller** [comment]
Just made a package for this! https://www.npmjs.com/package/next-optimized-classnames

**Reply by IRediTOTO** [reply]
Sorry but i just tried it on default nextjs, run build but nothing happen. How can this work? Can I use with tailwindcss?

**Reply by kenmueller** [reply]
What does your `next.config.js` look like?

**Reply by IRediTOTO** [reply]
Just simple module.exports = require('next-optimized-classnames')()

**Reply by kenmueller** [reply]
What does your code look like? Make sure you're using CSS modules. Tailwind won't work because it doesn't use CSS modules.

**Reply by salyadav** [reply]
I am using styled-components with nextjs here. Doesnt work for that either. 
The out directory has classnames like - `class="Header__Container-cd8ya4-1 joUQLx"`
Which is happening here-
https://github.com/vercel/next.js/blob/0af3b526408bae26d6b3f8cab75c4229998bf7cb/packages/next/build/webpack/config/blocks/css/loaders/getCssModuleLocalIdent.ts#L7

Looked into your code, which replaces this function basically. But unable to debug as to why its not working even in case of next with styled component.
